Expire deprecations in widgets and keyword only arguments for Selectors (#24254)
matplotlib/matplotlib: doc/api/next_api_changes/removals/24254-OG.rst

- The ``interactive`` attribute has been privatized and removed.
- The *marker_props* argument is removed, use *handle_props* instead.
- The *maxdist* argument is removed, use *grab_range* instead.
- The *rectprops* argument is removed, use *props* instead.
- The ``rectprops`` attribute has been privatized and removed.
- The ``state`` attribute has been privatized and removed.
- The ``to_draw`` attribute has been privatized and removed.

PolygonSelector
...............

- The *line* attribute is removed. If you want to change the selector artist
  properties, use the ``set_props`` or ``set_handle_props`` methods.
- The *lineprops* argument is removed, use *props* instead.
- The *markerprops* argument is removed, use *handle_props* instead.
- The *maxdist* argument and attribute is removed, use *grab_range* instead.
- The *vertex_select_radius* argument and attribute is removed, use
  *grab_range* instead.

SpanSelector
............

- The ``active_handle`` attribute has been privatized and removed.
- The ``eventpress`` attribute has been privatized and removed.
- The ``eventrelease`` attribute has been privatized and removed.

> - The *maxdist* argument and attribute is removed, use *grab_range* instead.

Reviewer: AFAICS SpanSelector did not have *maxdist*

Author: No idea. Straight C&P.

Reviewer 2: Looks like this came in via 00444e4ff3719c1ca0760e6e804d030e21f3f781 and we have been propogating it along.

> - The *maxdist* argument and attribute is removed, use *grab_range* instead.

Reviewer 2: ```suggestion
```

- The ``pressv`` attribute has been privatized and removed.
- The ``prev`` attribute has been privatized and removed.
- The ``rect`` attribute has been privatized and removed.

> - The *rectprops* argument is removed, use *props* instead.

Reviewer: ```suggestion
- The *rectprops* parameter has been renamed to *props*.
```
Note that there is a distinction "A parameter is the variable which is part of the method's signature (method declaration). An argument is an expression used when calling the method.", although that is not always used consistently. But parameter is more correct here, and it's also what we call the respective section in the docs.

Author: Again, C&P.

- The ``rectprops`` attribute has been privatized and removed.

> - The *span_stays* argument is removed, use the *interactive* argument
>   instead.

Reviewer: ```suggestion
- The *span_stays* parameter has been renamed to *interactive*.
```

- The ``span_stays`` attribute has been privatized and removed.
- The ``state`` attribute has been privatized and removed.

LassoSelector
.............

- The *lineprops* argument is removed, use *props* instead.
- The ``onpress`` and ``onrelease`` methods are removed. They are straight

>   aliases for ``press`` and ``release``.

Reviewer: Removal of `TextBox.DIST_FROM_LEFT` is missing.

Author: FWIW this is all just copied and pasted from the 3.5 release notes and deprecated replaced with removed (more or less).

But I will of course edit accordingly.

Reviewer 2: Looks like this was deprecated in bd9ee5e98616d96ab96fbef0d76d185bdf681cdd but we did not get a API change note with it.

Author: Should I update this to be deprecated in 3.7?

Reviewer 2: It has been warning, but was not documented.  The conservative thing to do is put it back, document it removed in 3.7 and drop it in 3.8, but I'm inclined to be a bit YOLO here and just document that it is removed in 3.7 as this is a very niche feature...
